mounter/exec.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mounter
+
+import "k8s.io/utils/exec"
+
+// NewOsExec returns a new Exec interface implemenation based on utils/exec
+func NewOsExec() Exec {
+	return &osExec{}
+}
+
+// Real implementation of Exec interface that uses simple utils.Exec
+type osExec struct{}
+
+var _ Exec = &osExec{}
+
+func (e *osExec) Run(cmd string, args ...string) ([]byte, error) {
+	exe := exec.New()
+	return exe.Command(cmd, args...).CombinedOutput()
+}
+
+// NewFakeExec returns a new FakeExec
+func NewFakeExec(run runHook) *FakeExec {
```

Comment:
@msau42 I took a stab at creating a `testing` subdir and putting the fakes in there, and immediately hit a snag. I want to get your opinion before I proceed with a solution...

We have three "fakes" -- `FakeMounter`, `FakeHostUtil`, and `FakeExec`. Two have them have implementations that reference types found in `mount`, so it has to be imported. However, a lot of the tests in `mount` use these fakes -- so moving the fakes into a new package creates an import cycle. The normal solution in this case, I believe, would be to create a `mount/types` package, and have them both import that. This will have a pretty large ripple effect, so I want to get your opinion on that before I try it out.

Even once that problem is solved, there is a second one. There are two places in the fakes that call functions found in `mount`, one of which isn't exported. we could export it, but end up with the import cycle again... I think  the way around this is to implement whatever functionality is needed in the fake directly -- it can probably even be simpler.  I think the re-use of functions there was out of convenience. but if we want a real "fake" that is separate, it can't re-use things from `mount`.  (well, types aside, see problem 1).

Reply:
Oh, and to answer Tim's question on whether `NewFakeExec` has a lot of callers -- it does. It is used by several volume plugins.

Reply:
Just randomly searching around for other fakes, can you see how pkg/kubelet/container/testing.FakeRuntime is solving this import cycle?

Reply:
Yeah I had looked at a bunch of those for inspiration. It looks like they just took the test that imports the fakes and made it a different package...

https://github.com/kubernetes/kubernetes/blob/acaac181dc8f4c16ed0f5e47f687a6475cae49f3/pkg/kubelet/container/runtime_cache_test.go#L17

That's simple enough. :)

Reply:
https://github.com/kubernetes/kubernetes/pull/81423